scripts/generator/graphql_generator/csharp/Cart.cs.erb

```diff
@@ -211,6 +225,35 @@ namespace <%= namespace %> {
             });
         }
 
+        public void SetFinalCheckoutFields(string email, ShippingFields? shippingFields, CompletionCallback callback) {
+            MutationQuery query = new MutationQuery();
+
+            DefaultQueries.checkout.EmailUpdate(query, CurrentCheckout.id(), email);
+
+            if (shippingFields.HasValue) {
+                DefaultQueries.checkout.ShippingAddressUpdate(query, CurrentCheckout.id(), shippingFields.Value.ShippingAddress);
+                DefaultQueries.checkout.ShippingLineUpdate(query, CurrentCheckout.id(), shippingFields.Value.ShippingIdentifier);
+            }
+
+            Client.Mutation(query, (Mutation response, ShopifyError error) => {
+                if (error != null) {
+                    callback(error);
+                } else {
+                    var userErrors = response.checkoutShippingAddressUpdate().userErrors();
+                    if (shippingFields.HasValue) {
+                        userErrors.AddRange(response.checkoutShippingLineUpdate().userErrors());
+                        userErrors.AddRange(response.checkoutEmailUpdate().userErrors());
+                    }
+
+                    if (UpdateState(response.checkoutEmailUpdate().checkout(), userErrors)) {
+                        PollCheckoutAndUpdate(PollCheckoutAvailableShippingRatesReady, callback);
```

Comment:
`PollCheckoutAndUpdate(PollCheckoutReady, callback)` instead because we want to wait for `Checkout.ready()`